### DISCUSSION TITLE
How to enable LaTeX preview in org, but with tectonic

### DISCUSSION THREAD

**Discussion by osvarcha** [tectonic-typesetting/tectonic#1098] [Q&A] [answered]
Hello, hello, I recently saw the documentation and the way of working with tectonic with auctex, but what I'm trying to do now is reproduce the equations or make a preview in Org-Mode just as the page [here](https://orgmode.org/worg/org-tutorials/org-latex-preview.html) shows, there is an expert in emacs -fu that you could guide me which variables I should manipulate so that I can visualize the equations or in general enrich my texts in org format with LaTex with the help of tectonic, because by default Org expects you to have a TeX distribution installed :c

**Comment by nanjj** [comment, marked as answer]
Add process name `tectonic` into `org-preview-latex-process-alist`:
```
  (tectonic :programs ("tectonic" "convert") 
               :description "pdf > png"
               :message "you need install the programs: tectonic and imagemagick."
               :image-input-type "pdf" 
               :image-output-type "png"
               :image-size-adjust (1.0 . 1.0) 
               :latex-compiler
                 ("tectonic -Z shell-escape-cwd=%o --outfmt pdf --outdir %o %f")
               :image-converter
               ("convert -density %D -trim -antialias %f -quality 300 %O"))
```
and customize `org-preview-latex-default-process` to `tectonic`.

You need to make sure both tectonic and imagemagick installed.

If you org files are in a soft linked directory you may meet the ralative path not correct issue, you can solve it via like to replace `%f` in  `latex-compiler` with `$(realpath --relative-to=$(pwd) %o%b.tex)` , and to replace %f in  `image-converter` with `$(realpath --relative-to=$(pwd) %o%b.pdf)` .

**Reply by fpdotmonkey** [reply]
For anyone that got confused what you're supposed to do here, this is what you want.

```lisp
(add-to-list 'org-preview-latex-process-alist
	     '(tectonic :programs ("tectonic" "convert") 
			:description "pdf > png"
			:message "you need install the programs: tectonic and imagemagick."
			:image-input-type "pdf" 
			:image-output-type "png"
			:image-size-adjust (1.0 . 1.0) 
			:latex-compiler
			("tectonic -Z shell-escape-cwd=%o --outfmt pdf --outdir %o %f")
			:image-converter
			("convert -density %D -trim -antialias %f -quality 300 %O")))
(setq org-preview-latex-default-process 'tectonic)
```
Additionally, if you get errors like I did about the pipe `|'inkscape' -V` not being allowed, you're probably using the svg package and you need to change its use to `\usepackage[inkscapeversion=1]{svg}`.

**Comment by osvarcha** [comment]
Muchas gracias me ayudo bastante tu configuración, estaba buscando la manera de obtener mis apuntes con termux, mas emacs y tectonic ahora esto es una maravilla.
